src/main/java/org/sonatype/repository/vgo/internal/hosted/VgoHostedFacetImpl.java

```diff
@@ -0,0 +1,239 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.impl.io.EmptyInputStream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.repository.view.Payload.UNKNOWN_SIZE;
+import static org.sonatype.repository.vgo.VgoAssetKind.VGO_MODULE;
+import static org.sonatype.repository.vgo.internal.util.VgoDataAccess.HASH_ALGORITHMS;
+
+/**
+ * Vgo hosted implementation
+ *
+ * @since 0.0.2
+ */
+@Named
+public class VgoHostedFacetImpl
+    extends FacetSupport
+  implements VgoHostedFacet
+{
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private final VgoDataAccess vgoDataAccess;
+
+  @Override
+  protected void doInit(final Configuration configuration) throws Exception {
+    super.doInit(configuration);
+  }
+
+  @Inject
+  public VgoHostedFacetImpl(final VgoDataAccess vgoDataAccess) {
+    this.vgoDataAccess = checkNotNull(vgoDataAccess);
+  }
+
+  @Override
+  @Nullable
+  @Transactional
+  public Content getInfo(final String path,
+                         final VgoAttributes vgoAttributes) {
+    checkNotNull(path);
+    String newPath = path.replaceAll("\\.info", "\\.zip");
+
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Asset asset = vgoDataAccess.findAsset(tx, tx.findBucket(getRepository()), newPath);
+    if (asset == null) {
+      return null;
+    }
+
+    StreamPayload streamPayload = new StreamPayload(
+        () -> doGetInfo(asset, vgoAttributes),
+        UNKNOWN_SIZE,
+        ContentTypes.APPLICATION_JSON);
+    return new Content(streamPayload);
+  }
+
+  @Transactional
+  @Override
+  public Content getList(final String module) {
+    checkNotNull(module);
+
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Iterable<Asset> assetsForModule = vgoDataAccess.findPackageAssetsForModule(tx, getRepository(), module);
+
+    String listOfVersions = StreamSupport.stream(assetsForModule.spliterator(), false)
+        .map(asset -> asset.name())
+        .map(name -> name.split("/@v/")[1])
+        .map((name -> name.replaceAll(".zip", "")))
+        .collect(Collectors.joining("\n"));
+
+    return new Content(
+      new StreamPayload(
+          new InputStreamSupplier() {
+            @Nonnull
+            @Override
+            public InputStream get() {
+              return new ByteArrayInputStream(listOfVersions.getBytes());
+            }
+          },
+          UNKNOWN_SIZE,
+          ContentTypes.TEXT_PLAIN
+      )
+    );
+  }
+
+  private InputStream doGetInfo(final Asset asset, final VgoAttributes vgoAttributes) {
+    VgoInfo vgoInfo = new VgoInfo(vgoAttributes.getVersion(), asset.blobCreated().toString());
+    try {
+      String info = MAPPER.writeValueAsString(vgoInfo);
+      return new ByteArrayInputStream(info.getBytes());
+    }
+    catch (JsonProcessingException e) {
+      log.warn(String.format("Unable to convert %s to json", vgoInfo.toString()));
```

Comment:
If this fails we probably want the stack trace.

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-vgo/pull/5/commits/b35a488a192dee3301aa70e1f7ede0b9ab783fd9